### PR TITLE
Getting project commits

### LINKED ISSUE #19: Send commits info to collector
In the release information, you must also send a field with information about commits made since the last release for the suspected commits feature to work

### PR DESCRIPTION
**Pull request by GeekaN2**
Adds options for collecting commits and sending them to the collector along with source maps

Resolves #19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,13 @@ Plugin options:
 | `release` | no | Unique identifier of the release. By default it will be Webpack's compilation hash. You **should** pass this identifier to the [Javascript Catcher](https://github.com/codex-team/hawk.javascript) on initialization through the `release` option |
 | `releaseInfoFile` | no | The path where `release.json` file will be created. By default, it will be got from Webpack `output.path` option. You can pass `false` to prevent creation of this file (can be useful, if you store and pass release id manually) |
 | `removeSourceMaps` | no | Should the plugin to remove emitted source map files. Default is `true`. |
+| `getCommits` | no | Should the plugin try to get commits from the repository. Default is `false` |
+| `gitOptions` | no | Options for getting the commits |
```
Comment (neSpecc) [inline README.md:39]: two options for a single reason. It's better to leave one of them. For example:

```
{
   // ...
   commits: { // also can be false
     repo: '',
     count: 5
   }
 }
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hawk.so/webpack-plugin",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Comment (neSpecc) [inline package.json:3]:
```suggestion
  "version": "0.2.0",
```

File: src/index.js [hunk 3]
```diff
@@ -13,6 +14,13 @@ const http = require('http');
  * @property {string} name - file name
  * @property {string} path - full path from the system root
  */
+/**
+ * @typedef {object} Commits
```
Comment (neSpecc) [inline src/index.js:18]:
```suggestion
 * @typedef {object} Commit
```

File: src/index.js [hunk 4]
```diff
@@ -132,6 +153,27 @@ class HawkWebpackPlugin {
...
+   * @return {Commits[]}
+   */
+  findCommits() {
+    const options = {
+      repo: this.gitOptions.repo,
+      number: this.gitOptions.number,
+      fields: ["hash", "subject", "authorEmail", "authorDate"],
+    }
+
+    const commits = gitlog(options);
```
Comment (neSpecc) [inline src/index.js:167]: do we need try-catch here?

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (GeekaN2) [reply README.md]: It is better not to rename `number` to  `count`. These settings are identical in the module, if the user makes a mistake, he will get the correct error

File: src/index.js (hunk 4 above)
Comment (GeekaN2) [reply src/index.js:170]: It is better to leave it, if the user specified the path incorrectly or something else happened, he could get user-friendly log

File: src/index.js [hunk 5]
```diff
@@ -48,12 +52,11 @@ class HawkWebpackPlugin {
     this.isHttps = this.collectorEndpoint.startsWith('https');
     this.requestTimeout = 50;
     this.removeSourceMaps = removeSourceMaps;
-    this.getCommits = getCommits;
-    this.gitOptions = {
+    this.commits = typeof commits === 'object' ? {
```
Comment (neSpecc) [inline src/index.js:55]:
```suggestion
    this.commits = typeof commits !== 'boolean' ? {
```

File: src/index.js [hunk 6]
```diff
@@ -159,9 +204,17 @@ class HawkWebpackPlugin {
           filename: map.name
         });
         body.append('release', releaseId);
-        body.append('commits', []);
 
-        // this.log(`Sending map [${map.name}] ...`)
+        if (this.commits) {
+          try {
+            const commits = this.findCommits() || [];
```
Comment (neSpecc) [inline src/index.js:210]: but what if there are no commits? or no git. `commits` does not checked for emptiness

**Review by nikmel2803 — APPROVED** [review]

File: src/index.js [hunk 7]
```diff
@@ -132,6 +156,27 @@ class HawkWebpackPlugin {
...
+      number: this.commits.number,
+      fields: ["hash", "subject", "authorEmail", "authorDate"],
+    }
+
+    const commits = gitlog(options);
+
+    return commits.map(commit => ({
+      hash: commit.hash,
+      title: commit.subject,
+      author: commit.authorEmail,
```
Comment (ilyamore88) [inline src/index.js:175]: Can we parse author's name here? And send in author field `Ilya Moroz <email@email.com>`

**Pull request merged**